firestore/transaction.go

```diff
@@ -116,11 +116,13 @@ func (c *Client) RunTransaction(ctx context.Context, f func(context.Context, *Tr
 	// TODO(jba): get backoff time from gRPC trailer metadata? See
 	// extractRetryDelay in https://code.googlesource.com/gocloud/+/master/spanner/retry.go.
 	for i := 0; i < t.maxAttempts; i++ {
+		t.ctx = trace.StartSpan(t.ctx, "cloud.google.com/go/firestore.Client.BeginTransaction")
```

Comment:
Is there a way to add these to the gapic methods for BeginTransaction and Commit perhaps? @benwhitehead

Reply:
I'm not sure. We'll likely have to follow up with someone familiar with the golang generator to see what they think. Do you know who that is off the top of your head?

Reply:
tbh I'm not really sure what is being asked for here 😅  my guess is that you'd like to see the ability to enable tracing at the _GAPIC method_ scope? This seems like something that could benefit others, I'm just not familiar with how it is used/configured/toggled, so I can't give you a real yes/no.

If y'all know of examples in other languages, that would be helpful. We should probably move this offline or to a [GitHub issue](https://github.com/googleapis/gapic-generator-go/issues) to discuss further. Maybe there is something we can do in GAX instead of the generator 🤷 . I think @codyoss knows about the tracing we enable in the repo, if any.

Reply:
@codyoss could you please take a look?

Reply:
I don't think this should be considered a blocker for this PR. I agree seems like a separate issue that we should track if we are interested.